common/av_log.c

```diff
@@ -32,12 +32,18 @@
 
 #include <libavutil/avutil.h>
 #include <libavutil/log.h>
+#include <libavutil/version.h>
 
 #include <libavcodec/avcodec.h>
+#include <libavcodec/version.h>
 #include <libavformat/avformat.h>
+#include <libavformat/version.h>
 #include <libswresample/swresample.h>
+#include <libswresample/version.h>
 #include <libswscale/swscale.h>
+#include <libswscale/version.h>
 #include <libavfilter/avfilter.h>
+#include <libavfilter/version.h>
 
 #if HAVE_LIBAVDEVICE
 #include <libavdevice/avdevice.h>
```

Comment:
what about avdevice?

Reply:
avdevice version isn't being checked as its addition was newer than the original list of checked versions. avdevice inclusion was originally added in dd3260185ac4a04277fc550ca2206cc29bf68bc6 just to be able to call `avdevice_register_all`. The version checks were not amended with this.